compoundicontextview/src/main/java/com/github/aakira/compoundicontextview/CompoundIconTextView.java

```diff
@@ -103,52 +113,42 @@ public CompoundIconTextView(Context context, @Nullable AttributeSet attrs, int d
      * @param resourceId Set the {@link CompoundIconTextView#UNDEFINED_RESOURCE} if you want clear icon.
      */
     public void setVectorDrawableLeft(@DrawableRes final int resourceId) {
-        if (resourceId == UNDEFINED_RESOURCE) {
-            drawables[INDEX_LEFT] = null;
-            drawableResIds[INDEX_LEFT] = UNDEFINED_RESOURCE;
-            updateIcons();
-        } else {
-            setVectorDrawable(INDEX_LEFT, resourceId);
-        }
```

Comment:
These logic is same in several method so I moved it to `setVectorDrawable()` method.